### PR TITLE
[W5][W17-4] Kevin Chan

### PR DESCRIPTION
**Pull request by kev-inc**

### CODE REVIEW

File: src/seedu/addressbook/commands/FilterCommand.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FilterCommand extends Command {
+  public static final String COMMAND_WORD = "filter";
+  public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all person whose names begin with the specified alphabet.\n"
```
Comment (yunjun199321) [inline src/seedu/addressbook/commands/FilterCommand.java:10]: This code seems too long. It may exceed 120 characters in one line.

**Comment by yunjun199321** [comment]
Good job Kavin.

File: src/seedu/addressbook/commands/FilterCommand.java (hunk 1 above)
Comment (adamwth) [inline src/seedu/addressbook/commands/FilterCommand.java:10]: What yunjun said - watch out for line character limit for adherence to coding standard

File: src/seedu/addressbook/parser/Parser.java [hunk 2]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (adamwth) [inline src/seedu/addressbook/parser/Parser.java:14]: Don't use wildcard imports. It is a coding standard violation. You may want to look into your IntelliJ configuration to see if it is doing so for you automatically (and turn it off if it is).

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -85,6 +78,9 @@ public Command parseCommand(String userInput) {
         case FindCommand.COMMAND_WORD:
             return prepareFind(arguments);
 
+        case FilterCommand.COMMAND_WORD:
```
Comment (adamwth) [inline src/seedu/addressbook/parser/Parser.java:81]: Good job for adhering to switch-case indentation

File: src/seedu/addressbook/parser/Parser.java [hunk 4]
```diff
@@ -248,5 +244,17 @@ private Command prepareFind(String args) {
         return new FindCommand(keywordSet);
     }
 
+    private Command prepareFilter(String args) {
+      final Matcher matcher = FILTER_ARGS_FORMAT.matcher(args.trim());
+      if (!matcher.matches()) {
+        return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                FilterCommand.MESSAGE_USAGE));
```
Comment (adamwth) [inline src/seedu/addressbook/parser/Parser.java:251]: Good job for indenting correctly for overly long lines

**Review by adamwth — APPROVED** [review]
Points to note:
- Good job for remembering to update docs
- Good job for updating I/O tests, but remember to update JUnit tests too
- Watch out for coding standard violations

**Pull request closed without merging**